### PR TITLE
Feature/dei 105 feedback on multiple classification rule

### PR DESCRIPTION
**Pull request by CindyvdVries**

### CODE REVIEW

File: decoimpact/data/parsers/parser_classification_rule.py [hunk 1]
```diff
@@ -114,257 +114,169 @@ def check_coverage(criteria_table, conditions={}, unique=True):
...
+        self, name, criteria, msgs, pre_warn, logger: ILogger
     ):
-        not_covered_values = [-_np.inf, _np.inf]
-        covered_numbers = []
+        range_criteria = list(map(self._convert_to_ranges, criteria))
+        sorted_range_criteria = sorted(range_criteria, key=lambda x: x[1])
+        sorted_range_criteria = sorted(sorted_range_criteria, key=lambda x: x[0])
 
-        all_criteria = [type_of_classification(val) for val in criteria]
         # Check if there are multiple larger or larger and equal comparison values are
```
Comment (wschoonveld) [inline decoimpact/data/parsers/parser_classification_rule.py:158]: larger = smaller?

File: decoimpact/data/parsers/parser_classification_rule.py [hunk 2]
```diff
@@ -54,3 +61,223 @@ def parse_dict(self, dictionary: Dict[str, Any], logger: ILogger) -> IRuleData:
...
+            return [float(val), float(val), "equal"]
+
+        elif type_of_classification(val) == "range":
+            start_range, end_range = str_range_to_list(val)
+            return [start_range, end_range, "equal"]
+
+        else:
+            return [float("-inf"), float("inf")]
+
+    def _validate_criteria_on_overlap_and_gaps(
```
Comment (wschoonveld) [inline decoimpact/data/parsers/parser_classification_rule.py:151]: please provide description for this function

File: decoimpact/data/parsers/parser_classification_rule.py [hunk 3]
```diff
@@ -54,3 +61,223 @@ def parse_dict(self, dictionary: Dict[str, Any], logger: ILogger) -> IRuleData:
...
+        self, name, criteria, msgs, pre_warn, logger: ILogger
+    ):
+        range_criteria = list(map(self._convert_to_ranges, criteria))
+        sorted_range_criteria = sorted(range_criteria, key=lambda x: x[1])
+        sorted_range_criteria = sorted(sorted_range_criteria, key=lambda x: x[0])
```
Comment (wschoonveld) [inline decoimpact/data/parsers/parser_classification_rule.py:156]: what does this do?

File: decoimpact/data/parsers/parser_classification_rule.py [hunk 4]
```diff
@@ -54,3 +61,223 @@ def parse_dict(self, dictionary: Dict[str, Any], logger: ILogger) -> IRuleData:
             output_variable_name,
             description,
         )
+
+    def _validate_table_coverage(self, crit_table, logger: ILogger):
+        msgs = []
+        criteria_table = crit_table.copy()
+        del criteria_table["output"]
+
+        def check_coverage(criteria_table, conditions={}, unique=True):
```
Comment (wschoonveld) [inline decoimpact/data/parsers/parser_classification_rule.py:70]: if I understand correctly, this is done via a recursive function to sort all criteria (and then validate them); correct? 
Maybe rename then or add this to the description?

**Review by wschoonveld — COMMENTED** [review]
please change/add some comments to clarify

**Review by mKlapwijk — APPROVED** [review]

**Pull request merged**